swe/solver.py

```diff
@@ -474,14 +476,54 @@ def solve_step(self, adjoint=False):
         self.solver_obj.iterate(update_forcings=self.op.get_update_forcings(self.solver_obj),
                                 export_func=self.op.get_export_func(self.solver_obj))
         self.solution = self.solver_obj.fields.solution_2d
-
+        
+        old_mesh = Mesh(Function(self.mesh.coordinates))                
+        P1DG = FunctionSpace(old_mesh, "DG", 1)
+        P1 = FunctionSpace(old_mesh, "CG", 1)    
+        
+        solution_bathymetry = self.solver_obj.fields.bathymetry_2d.copy(deepcopy = True)
+        self.op.solution_old_bathymetry = Function(P1).project(solution_bathymetry)
+        
+        if self.op.solve_tracer:
+            solution_tracer = self.solver_obj.fields.tracer_2d.copy(deepcopy = True)
+            self.op.solution_old_tracer = Function(P1DG).project(solution_tracer)
+            
+            
     def setup_solver(self):
         if not hasattr(self, 'remesh_step'):
             self.remesh_step = 0
         op = self.op
-        self.solver_obj = solver2d.FlowSolver2d(self.mesh, self.fields['bathymetry'])
+        
+        if hasattr(self.op, "solution_old_bathymetry"):
+            self.op.bathymetry = Function(self.P1).project(self.op.solution_old_bathymetry)
+        else:
+            self.op.bathymetry = self.op.set_bathymetry(self.P1)
+        
+        self.solver_obj = solver2d.FlowSolver2d(self.mesh, self.op.bathymetry)
+
```

Comment:
Can we not just use `self.fields['bathymetry'] = ...` instead of `self.op.bathymetry = ...`?